staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -72,10 +74,12 @@ func newETCD3HealthCheck(c storagebackend.Config) (func() error, error) {
 		client := clientValue.Load().(*clientv3.Client)
 		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
 		defer cancel()
-		if _, err := client.Cluster.MemberList(ctx); err != nil {
-			return fmt.Errorf("error listing etcd members: %v", err)
+		// See https://github.com/etcd-io/etcd/blob/master/etcdctl/ctlv3/command/ep_command.go#L118
+		_, err := client.Get(ctx, "health")
+		if err == nil || err == rpctypes.ErrPermissionDenied {
+			return nil
```

Comment:
I don't think we should report healthy on `rpctypes.ErrPermissionDenied`.

Reply:
I think you're right, I took this from etcd's healthcheck but it doesn't make a whole lot of sense for K8s which needs API access in order to function